ansible/content-servers.yml

```diff
@@ -0,0 +1,197 @@
+          ssl_session_cache shared:MozSSL:10m;  # about 40000 sessions
+          ssl_session_tickets off;
+          ssl_protocols TLSv1.2 TLSv1.3;
+          ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+          ssl_prefer_server_ciphers off;
+
+          {{ nginx_shared_config }}
+
+  pre_tasks:
+    - name: Update & upgrade & cleanup apt.
+      apt:
+        update_cache: true
+        upgrade: safe
+        autoremove: yes
+        autoclean: yes
+        cache_valid_time: 86400  # a day is plenty
+      changed_when: false
+
+    - name: Install apt https support & cron.
+      apt:
+        name:
+          - apt-transport-https
+          - cron  # required for certbot
+        state: present
+
+  roles:
+    - geerlingguy.certbot
+    - geerlingguy.nginx
+
+  tasks:
+    - name: Flush handlers in case any configs have changed.
+      meta: flush_handlers
+
+    # Tests basic nginx
+    - name: Test secure connection to SSL domain.
+      uri:
+        url: https://{{ inventory_hostname }}/healthz
+        status_code: 200
+      delegate_to: localhost
+      become: false
+
+    # Tests base path
+    - name: Test secure connection to SSL domain.
+      uri:
+        url: https://{{ inventory_hostname }}/
+        status_code: 404
+      delegate_to: localhost
+      become: false
+
+    - name: Install some helpful utilities.
+      apt:
+        name:
+          - bash-completion
+          - logrotate
+          - molly-guard
+          - rsync
+          - sshguard
+          - unattended-upgrades
+          - vim
+        state: present
+
+    - name: Copy sshd config.
+      copy:
+        src: files/sshd_config
+        dest: /etc/ssh/sshd_config
+      notify:
+        restart ssh
+
+    - name: Create sshguard blacklist db directory.
+      file:
+        path: /var/db/sshguard
+        state: directory
+
+    # By default, sshguard listens to the entire journal... which includes itself. It gets a bit confused. Add -t sshd to stop it being silly
+    - name: Configure sshguard properly.
+      lineinfile:
+        path: /etc/sshguard/sshguard.conf
+        regexp: "{{ item.regexp }}"
+        line: "{{ item.line }}"
+      with_items:
+        - regexp: "LOGREADER="
+          line: 'LOGREADER="LANG=C /bin/journalctl -afb -p info -n1 -t sshd -o cat SYSLOG_FACILITY=4 SYSLOG_FACILITY=10"'
+        - regexp: "BLACKLIST_FILE="
+          line: 'BLACKLIST_FILE=100:/var/db/sshguard/blacklist.db'
+      notify:
+        - restart sshguard
+
+    - name: Enable persistent systemd journal.
+      lineinfile:
+        path: /etc/systemd/journald.conf
+        regexp: "Storage="
+        line: "Storage=persistent"
+
+    - name: Add FQDN to /etc/hosts.
+      lineinfile:
+        path: /etc/hosts
+        regexp: "127.0.1.186" #0xBA
```

Comment:
Not sure what that comment is doing there? And what an odd localhost address .. did you make up that number? Or do you know why it is this number?